### PR TITLE
fix warning message in FitTrace when many bins masked

### PR DESCRIPTION
**Pull request by cshanahan1**
This PR consolidates warning messages that were previously produced for each individual masked column into one message reporting all masked columns, or that many columns were masked if there are more than 20.

### CODE REVIEW

File: specreduce/tracing.py [hunk 1]
```diff
@@ -346,6 +346,16 @@ def __post_init__(self):
                 # TODO: implement smoothing with provided width
                 y_bins[i] = ilum2[z_i.argmax()]
 
+        # warn about fully-masked bins (which, currently, means any bin with a single masked value)
+        if len(warn_bins) > 0:
+            plural = 's' if len(warn_bins) > 1 else ''
+            # if there are a ton of bins, we don't want to print them all out
+            print(warn_bins)
```
Comment (kecnry) [inline specreduce/tracing.py:353]:
```suggestion
```

File: specreduce/tracing.py [hunk 2]
```diff
@@ -346,6 +346,15 @@ def __post_init__(self):
...
                 y_bins[i] = ilum2[z_i.argmax()]
 
+        # warn about fully-masked bins (which, currently, means any bin with a single masked value)
+        if len(warn_bins) > 0:
+            plural = 's' if len(warn_bins) > 1 else ''
+            # if there are a ton of bins, we don't want to print them all out
+            if len(warn_bins) > 20:
+                warn_bins = warn_bins[0: 10] + ['...'] + [warn_bins[-1]]
+            warnings.warn(f"All pixels in bin{plural} {', '.join([str(x) for x in warn_bins])}"
+                          " are masked. Falling to trace value from all-bin fit.")
```
Comment (kecnry) [inline specreduce/tracing.py:356]: I know this was the previous message, but maybe we should try to clarify while we're editing this - is this more accurate?  (If so, the test would need updating as well).

```suggestion
                          " are masked. Falling back on trace value from all-bin fit.")
```

File: specreduce/tracing.py [hunk 3]
```diff
@@ -346,6 +346,15 @@ def __post_init__(self):
                 # TODO: implement smoothing with provided width
                 y_bins[i] = ilum2[z_i.argmax()]
 
+        # warn about fully-masked bins (which, currently, means any bin with a single masked value)
+        if len(warn_bins) > 0:
+            plural = 's' if len(warn_bins) > 1 else ''
+            # if there are a ton of bins, we don't want to print them all out
+            if len(warn_bins) > 20:
+                warn_bins = warn_bins[0: 10] + ['...'] + [warn_bins[-1]]
+            warnings.warn(f"All pixels in bin{plural} {', '.join([str(x) for x in warn_bins])}"
```
Comment (kecnry) [inline specreduce/tracing.py:355]: Could do the `plural` logic inline (and remove it from earlier)... might be more readable, but up to you.

```suggestion
            warnings.warn(f"All pixels in {'bins' if len(warn_bins) else 'bin'} {', '.join([str(x) for x in warn_bins])}"
```

**Review by kecnry — COMMENTED** [review]
Just a few small suggestions, but otherwise looks good to me once this is rebased to handle the conflicts.  Thanks!

File: specreduce/tracing.py (hunk 2 above)
Comment (cshanahan1) [reply specreduce/tracing.py]: done

**Review by kecnry — APPROVED** [review]

**Pull request merged**